### ISSUE TITLE
knife diff should prepend 'server' or 'local' to file paths

### ISSUE THREAD

**Issue by nogweii** [chef/chef#2269]
Given this knife diff header:

``` diff
diff --knife environments/production.json environments/production.json
--- environments/production.json        2014-10-21 10:25:26.370000424 -0700
+++ environments/production.json        2014-10-21 10:25:26.369000420 -0700
@@ -30,8 +30,8 @@
```

How does one know that the first file mentioned is the version stored on the server, while the second file mentioned is the copy stored locally? Thus, I propose:

``` diff
diff --knife server/environments/production.json local/environments/production.json
--- server/environments/production.json        2014-10-21 10:25:26.370000424 -0700
+++ local/environments/production.json        2014-10-21 10:25:26.369000420 -0700
@@ -30,8 +30,8 @@
```

**Comment by Romascopa** [comment]
I think a better use case would be to make it work like diff should work versus forcing client and server into the picture to work 100% that way. E.g., knife diff -client file1 -client file2 or knife diff -client file1 -server file2 or knife diff -server file1 -server file2

**Comment by tas50** [comment]
Thanks for submitting this issue. After review by the team, we've decided that this isn't something that we're going to work on ourselves. We receive dozens of new issues a month, which is more than our team can possibly work on. This means we must prioritize work and be honest about what we're able to get done as a team.  That doesn't mean that we think this is a bad idea or isn't a necessary change. It's just not something we'll likely get to given competing priorities. I've applied our `Design Proposal: Accepted. PRs Welcome` label to this issue and closed it out. If you or someone else would like to work on this issue we're open to pull requests. If you'd like to chat further about what that design might look reach out to us in our [#chef-infra-dev](https://chefcommunity.slack.com/archives/C55FCUGBS) slack channel.
